### PR TITLE
fix(devtools-proxy-support): handle special characters in SSH URL correctly COMPASS-8254

### PR DESCRIPTION
**Pull request by addaleax**
This is unfortunately very obvious in retrospect 😞

### CODE REVIEW

**Review by Anemy — APPROVED** [review]

File: packages/devtools-proxy-support/src/ssh.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ export class SSHAgent extends AgentBase implements AgentWithInitialize {
     this.logger.emit('ssh:establishing-conection', {
       host: sshConnectConfig.host,
       port: sshConnectConfig.port,
-      password: !!sshConnectConfig.passphrase,
+      password: !!sshConnectConfig.password,
```
Comment (lerouxb) [inline packages/devtools-proxy-support/src/ssh.ts:95]: 😭

**Review by lerouxb — APPROVED** [review]

File: packages/devtools-proxy-support/src/ssh.ts (hunk 1 above)
Comment (addaleax) [reply packages/devtools-proxy-support/src/ssh.ts:95]: To be fair, this bit is kinda unrelated, but still, yeah, not my smartest move 😅

Comment (lerouxb) [reply packages/devtools-proxy-support/src/ssh.ts:95]: Yeah I see now it is just in a logging event 👍

File: packages/devtools-proxy-support/src/ssh.ts [hunk 2]
```diff
@@ -78,10 +78,10 @@ export class SSHAgent extends AgentBase implements AgentWithInitialize {
     const sshConnectConfig: ConnectConfig = {
       readyTimeout: 20000,
       keepaliveInterval: 20000,
-      host: this.url.hostname,
+      host: decodeURIComponent(this.url.hostname),
```
Comment (lerouxb) [inline packages/devtools-proxy-support/src/ssh.ts:81]: I can see how this would be easy to miss.

**Pull request merged**